week-2/mandatory/3-api/server.js

```diff
@@ -0,0 +1,382 @@
+      (error, result) => {
+        if (error) {
+          console.log("pg error code:", error.code);
+          res.status(500).json(serverError);
+        } else {
+          res.status(200).json(result.rows);
+        }
+      }
+    );
+  }
+});
+
+app.post("/customers", (req, res) => {
+  let name = req.body.name,
+    address = req.body.address,
+    city = req.body.city,
+    country = req.body.country;
+  let sql =
+    "INSERT INTO customers (name, address, city, country) " +
+    "VALUES ($1, $2, $3, $4) returning id";
+  let attributes = [name, address, city, country];
+
+  db.query(sql, attributes, (error, result) => {
+    if (error) {
+      console.log("pg error code:", error.code);
+      res.status(500).json(serverError);
+    } else {
+      res
+        .status(201)
+        .json(`A new customer is created with the ID ${result.rows[0].id}`);
+    }
+  });
+});
+
+app.post("/products", (req, res) => {
+  let product = req.body.productName;
+  let sql = "insert into products (product_name)" + "values ($1) returning id";
+  db.query(sql, [product], (error, result) => {
+    if (error) {
+      console.log("pg error code:", error.code);
+      res.status(500).json(serverError);
+    } else {
+      res.status(201).json(`Product added with the ID ${result.rows[0].id}`);
+    }
+  });
+});
+
+app.post("/availability", (req, res) => {
+  let price = parseInt(req.body.price),
+    product = parseInt(req.body.product_id),
+    supplier = parseInt(req.body.supplier_id);
+  let sql_prod = "select id from products where id = $1 ",
+    sql_supp = " select id from suppliers where id  = $1 ",
+    sql_insert =
+      "insert into product_availability (prod_id,supp_id,unit_price)" +
+      "values ($1,$2,$3)";
+  if (price > 0) {
+    db.query(sql_supp, [supplier], (error, result) => {
+      if (error) {
+        console.log("pg error code:", error.code);
+        res.status(500).json(serverError);
+      } else if (result.rows.length > 0) {
+        db.query(sql_prod, [product], (error, result) => {
+          if (error) {
+            console.log("pg error code:", error.code);
+            res.status(500).json(serverError);
+          } else if (result.rows.length > 0) {
+            db.query(sql_insert, [product, supplier, price], (error) => {
+              if (error) {
+                console.log("pg error code:", error.code);
+                res.status(500).json(serverError);
+              } else {
+                res.status(201).json("Product price added.");
+              }
+            });
+          } else {
+            res.status(404).json("Can't find a product with this ID");
+          }
+        });
+      } else {
+        res.status(404).json("Can't find a supplier with this ID");
+      }
+    });
+  } else {
+    res.status(400).json("The price needs to be larger than 0");
+  }
+});
```

Comment:
There are a lot of nested if statements here, which makes it less readable. You could reduce this if you reversed the conditions of some of the if clauses and added returns, i.e.
```
if (price < 0) {
res.status(400).json("The price needs to be larger than 0");
return;
}
```
However there is a school of thought that having multiple returns in a function is a bad thing. So your code isn't wrong, but it's good to start thinking about different ways of organising your code and which you prefer and why.